JollyJumpers/JollyJumpersC/JollyJumpersC/main.c

```diff
@@ -3,25 +3,42 @@
 #endif
 
 #include <stdio.h>
+#include <stdbool.h>
 #define MAXSIZE 3000
 main()
 {
-	int n, i, j, a, Var[MAXSIZE], Abs[MAXSIZE];
+	//
+	// Dave:  The Abs array can be of type bool -- it might also be better named since it has the same name as the abs function
+	// Michelle: done
+	//
+	int numInts, i, a, Var[MAXSIZE];
+	bool AbsValue[MAXSIZE] = { 0 };
 
-	while (scanf("%d", &n) == 1) {				//while not end of file
-		for (i = 0; i < n; i++)					// for the nubmer of items on the line
+	while (scanf("%d", &numInts) == 1) {				//while not end of file
+														//
+														// Dave:  You shouldn't need a pass to read in the numbers and a second pass to calculate the difference.
+														//        Reading in the values and calculating the difference can be done in one pass.
+														// Will fix in the c++ & C# versions.
+														//
+		for (i = 0; i < numInts; i++)					// for the number of items on the line
 		{
-			scanf("%d", &Var[i]);				//shove that number into the varriable array
-			Abs[i] = 0;							//set the memory for it in the absolute array to 0 - no one likes bad memory bits
+			scanf("%d", &Var[i]);				//shove that number into the variable array
+			AbsValue[i] = 0;							//set the memory for it in the absolute array to 0 - no one likes bad memory bits
 		}
-		j = n - 1;	// number of items to go into the abs value array
 
-		for (i = 0; i < j; i++) {
-			Abs[abs(Var[i] - Var[i + 1])] = 1;	// loop through the var array, store a 1 at the location of the absolute difference between the values
+
+		for (i = 0; i < (numInts - 1); i++) {
+			AbsValue[abs(Var[i] - Var[i + 1])] = 1;	// loop through the var array, store a 1 at the location of the absolute difference between the values
```

Comment:
Is it possible for abs(Var[i] - Var[i+1]) to exceed the bounds of the AbsValue array?